cmd/check-gh-automation/main.go

```diff
@@ -206,8 +220,20 @@ func checkRepos(repos []string, bots []string, appName string, ignore sets.Set[s
 			}
 
 			// If branch protection is configured, verify admin access for the hardcoded admin bot
+			// also branch protection should be used only on public repos or paid plans
 			if branchProtectionEnabled {
 				logger.Infof("Branch protection is enabled for %s/%s", org, repo)
+
+				if err != nil {
+					logger.Errorf("Error gathering repository data in %s/%s: %v", org, repo, err)
+					return nil, fmt.Errorf("error gathering repository data in %s/%s: %w", org, repo, err)
+				}
+				if fullRepo.Private && fullOrg.Plan.Name == "free" {
```

Comment:
Just thought of this, but is it possible for a repo to have its own "plan"? We should verify this prior to merging...

Reply:
There is no "plan" property on "get a repository" api endpoint:
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository

Also, the [Github documentation](https://docs.github.com/en/get-started/learning-about-github/githubs-plans) says that is only available for accounts:
```GitHub offers free and paid plans for storing and collaborating on code. Some plans are available only to personal accounts, while other plans are available only to organization and enterprise accounts.```